backend/src/main/java/com/rota/api/StaffService.java

```diff
@@ -34,7 +38,7 @@
     return engagementRepository.findByStaffId(staffId).stream()
         .filter(engagement ->
             (engagement.getStart().equals(start)
-                || engagement.getStart().isAfter(startTime))
+                || !engagement.getStart().isBefore(startTime))
```

Comment:
With this `engagement.getStart().equals(start)` is no longer needed.
These two lines can be written simply as `!engagement.getStart().isBefore(startTime))`